src/context/analysisResultStateSlice.ts

```diff
@@ -0,0 +1,311 @@
+    return LayerDefinitions[this.baselineLayerId] as NSOLayerProps;
+  }
+}
+
+export type TableRow = {
+  localName: string;
+  name: string;
+  baselineValue: DataRecord['value'];
+  coordinates?: Position;
+} & { [k in AggregationOperations]?: number };
+
+const initialState: AnalysisResultState = {
+  isLoading: false,
+  isMapLayerActive: true, // TODO toggle via UI
+};
+
+function generateTableFromApiData(
+  statistic: AggregationOperations,
+  aggregateData: AsyncReturnType<typeof fetchApiData>, // data from api
+  // admin layer, both props and data. Aimed to closely match LayerData<BoundaryLayerProps>
+  {
+    layer: adminLayer,
+    data: adminLayerData,
+  }: { layer: BoundaryLayerProps; data: BoundaryLayerData },
+  // baseline layer, both props and data
+  {
+    layer: { adminLevel },
+    data: { layerData },
+  }: { layer: NSOLayerProps; data: NSOLayerData },
+): TableRow[] {
+  // find the key that will let us reference the names of the bounding boxes.
+  const adminLevelName =
+    adminLayer.adminLevelNames[adminLevel - 1] || adminLayer.adminLevelNames[0];
+  // for local name too.
+  const adminLevelLocalName =
+    adminLayer.adminLevelLocalNames[adminLevel - 1] ||
+    adminLayer.adminLevelLocalNames[0];
+
+  return aggregateData.map((row: any) => {
+    // find feature from admin boundaries that closely matches this api row.
+    // once we find it we can get the corresponding local name.
+    const featureBoundary = find(
+      adminLayerData.features,
+      (feature: any) =>
+        feature.properties[adminLevelName] === row[adminLevelName],
+    );
+
+    const name: string =
+      featureBoundary?.properties?.[adminLevelName] || 'No Name';
+    const localName: string =
+      featureBoundary?.properties?.[adminLevelLocalName] || 'No Name';
+
+    const baselineValue =
+      layerData.find(({ adminKey }) => {
+        return (featureBoundary?.properties?.[
+          adminLayer.adminCode
+        ] as string).startsWith(adminKey);
+      })?.value || 'No Data';
+
+    const tableRow: TableRow = {
+      name,
+      localName,
+      [statistic]: get(row, statistic, 0),
+      baselineValue,
+      coordinates: (featureBoundary?.geometry as any).coordinates[0][0][0], // TODO likely will not keep
+    };
+    return tableRow;
+  });
+}
+
+export type AnalysisDispatchParams = {
+  baselineLayer: NSOLayerProps;
+  hazardLayer: WMSLayerProps;
+  extent: Extent;
+  threshold: ThresholdDefinition;
+  date: ReturnType<Date['getTime']>; // just a hint to developers that we give a date number here, not just any number
+  statistic: AggregationOperations; // we might have to deviate from this if analysis accepts more than what this enum provides
+};
+const apiUrl = 'https://prism-api.ovio.org/stats'; // TODO both needs to be stored somewhere
+// TODO - we need to use the public version generated by the app itself
+const adminJson =
+  'https://prism-admin-boundaries.s3.us-east-2.amazonaws.com/mng_admin_boundaries.json';
```

Comment:
@SimplyBallistic actually we should be using the administrative file from the frontend as this will change for each country. So you should be sending something like "public_url/${layers.admin_boundaries.path}" instead

Reply:
@ericboucher we indeed are using the local copy of admin_boundaries, however for development, we use this link instead. I'm currently using `NODE_ENV==='production'` on line 185 to decide which to use.

Reply:
...Mentioned on #98